compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1970,57 +1970,128 @@ setTransformStrategyRootConfig(mlir::FunctionOpInterface entryPointFn,
   return success();
 }
 
-/// Sets the lowering configuration for a generic op implementing a
-/// transposition to use CPUDoubleTilingExpert pipeline.
-static LogicalResult
-setTransposeLikeOpRootConfig(mlir::FunctionOpInterface entryPointFn,
-                             linalg::GenericOp genericOp,
-                             const LinalgOpInfo &linalgOpInfo,
-                             const TargetMLTransformInfo &targetMLTransInfo) {
-  assert(!getLoweringConfig(genericOp) &&
-         "expected lowering_config is not set");
-
-  LLVM_DEBUG(KD_DBGS() << "Setting transpose-like op root configuration\n");
-
-  auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(entryPointFn);
+/// Utility to return the transpose vector `sizes` for X86. Empty `sizes` on
+/// return indicates failure.
+static void getTransposeX86VectorSizes(
+    linalg::GenericOp genericOp, IREE::HAL::ExecutableTargetAttr targetAttr,
+    SmallVectorImpl<int64_t> &minTileSizes, SmallVectorImpl<int64_t> &sizes) {
```

Comment:
I think the type of minTileSizes can be `ArrayRef<int64_t>`. We don't expect that the it would be modified. At the last part, you can do something like:

```
sizes.assign(minTileSizes);
std::replace_if(sizes.begin(), sizes...
```